tests/test_scalars.py

```diff
@@ -0,0 +1,663 @@
+
+    check(assert_type(td.ceil("D"), pd.Timedelta), pd.Timedelta)
+    check(assert_type(td.floor(Day()), pd.Timedelta), pd.Timedelta)
+    check(assert_type(td.isoformat(), str), str)
+    check(assert_type(td.round("s"), pd.Timedelta), pd.Timedelta)
+    check(assert_type(td.to_numpy(), np.timedelta64), np.timedelta64)
+    check(assert_type(td.to_pytimedelta(), dt.timedelta), dt.timedelta)
+    check(assert_type(td.to_timedelta64(), np.timedelta64), np.timedelta64)
+    check(assert_type(td.total_seconds(), float), float)
+    check(assert_type(td.view(np.int64), object), np.int64)
+    check(assert_type(td.view("i8"), object), np.int64)
+
+
+def test_timedelta_add_sub() -> None:
+    td = pd.Timedelta("1 day")
+
+    ndarray_td64: npt.NDArray[np.timedelta64] = np.array(
+        [1, 2, 3], dtype="timedelta64[D]"
+    )
+    ndarray_dt64: npt.NDArray[np.datetime64] = np.array(
+        [1, 2, 3], dtype="datetime64[D]"
+    )
+    as1 = pd.Period("2012-01-01", freq="D")
+    as2 = pd.Timestamp("2012-01-01")
+    as3 = dt.datetime(2012, 1, 1)
+    as4 = dt.date(2012, 1, 1)
+    as5 = np.datetime64(1, "ns")
+    as6 = dt.timedelta(days=1)
+    as7 = np.timedelta64(1, "D")
+    as8 = pd.TimedeltaIndex([td])
+    as9 = pd.Series([td])
+    as10 = pd.period_range("2012-01-01", periods=3, freq="D")
+    as11 = pd.date_range("2012-01-01", periods=3)
+    as12 = ndarray_td64
+    as13 = ndarray_dt64
+    as14 = pd.NaT
+
+    check(assert_type(td + td, pd.Timedelta), pd.Timedelta)
+    check(assert_type(td + as1, pd.Period), pd.Period)
+    check(assert_type(td + as2, pd.Timestamp), pd.Timestamp)
+    check(assert_type(td + as3, pd.Timestamp), pd.Timestamp)
+    check(assert_type(td + as4, dt.date), dt.date)
+    check(assert_type(td + as5, pd.Timestamp), pd.Timestamp)
+    check(assert_type(td + as6, pd.Timedelta), pd.Timedelta)
+    check(assert_type(td + as7, pd.Timedelta), pd.Timedelta)
+    check(assert_type(td + as8, pd.TimedeltaIndex), pd.TimedeltaIndex)
+    check(assert_type(td + as9, pd.Series), pd.Series)
+    check(assert_type(td + as10, pd.PeriodIndex), pd.PeriodIndex)
+    check(assert_type(td + as11, pd.DatetimeIndex), pd.DatetimeIndex)
+    check(assert_type(td + as12, npt.NDArray[np.timedelta64]), np.ndarray)
+    # pyright has trouble with timedelta64 and datetime64
+    check(
+        assert_type(
+            td + as13,  # pyright: ignore[reportGeneralTypeIssues]
+            npt.NDArray[np.datetime64],
+        ),
+        np.ndarray,
+    )
+    check(assert_type(td + as14, NaTType), NaTType)
+
+    check(assert_type(as1 + td, pd.Period), pd.Period)
+    check(assert_type(as2 + td, pd.Timestamp), pd.Timestamp)
+    check(assert_type(as3 + td, dt.datetime), dt.datetime)
+    check(assert_type(as4 + td, dt.date), dt.date)
+    check(assert_type(as5 + td, pd.Timestamp), pd.Timestamp)
+    # pyright is wrong here because as6 + td calls td.__radd__(as6),
+    # not timedelta.__add__
```

Comment:
So there is a rule here where `Timedelta.__radd__()` is called, but it can't be determined with static typing.  See https://github.com/microsoft/pyright/issues/4088

Maybe put a link to that issue.  `pyright` isn't wrong - it's that it can't be figured out in a static typing context.

Reply:
Added a note. I don't think the explanation given makes sense.  When you have a concrete `Timedelta`, you know that the method called will be `Timedelta.__radd__` since it has priority over `timedelta.__radd__`.

Reply:
Thanks.  Nothing we can rally do about it by the looks of things.